src/tests/mocks/issue.rs

```diff
@@ -0,0 +1,290 @@
+    action: IssueCommentEventAction,
+    issue: Issue,
+    comment: Comment,
+    repository: Repository,
+}
+
+impl IssueCommentEventPayload {
+    pub(crate) fn new(comment_body: &str) -> Self {
+        Self {
+            action: IssueCommentEventAction::Created,
+            issue: Issue::default(),
+            comment: Comment::new(comment_body),
+            repository: Repository::default(),
+        }
+    }
+}
+
+#[derive(Serialize)]
+#[serde(rename_all = "lowercase")]
+enum IssueCommentEventAction {
+    Created,
+}
+
+#[derive(Serialize)]
+struct Label {
+    id: u64,
+    node_id: String,
+    url: Url,
+    name: String,
+    color: String,
+    default: bool,
+}
+
+impl Default for Label {
+    fn default() -> Self {
+        Label {
+            id: 1,
+            node_id: "".to_string(),
+            url: "https://test.com".parse().unwrap(),
+            name: "test".to_string(),
+            color: "000000".to_string(),
+            default: false,
+        }
+    }
+}
+
+#[derive(Serialize)]
+struct PullRequest {
+    url: String,
+    id: u64,
+    title: String,
+    body: String,
+
+    /// The pull request number.  Note that GitHub's REST API
+    /// considers every pull-request an issue with the same number.
+    number: u64,
+
+    head: Box<Head>,
+    base: Box<Base>,
+}
+
+impl Default for PullRequest {
+    fn default() -> Self {
+        PullRequest {
+            url: "https://test.com".to_string(),
+            id: 1,
+            title: "test".to_string(),
+            body: "test".to_string(),
+            number: default_pr_number(),
+            head: Box::new(Head {
+                label: "test".to_string(),
+                ref_field: "test".to_string(),
+                sha: "test".to_string(),
+            }),
+            base: Box::new(Base {
+                ref_field: "test".to_string(),
+                sha: "test".to_string(),
+            }),
+        }
+    }
+}
+
+pub(super) async fn setup_pull_request_mock(mock_server: &MockServer) {
+    let repo = Repository::default();
+    Mock::given(method("GET"))
+        .and(path(format!(
+            "/repos/{}/{}/pulls/{}",
+            repo.owner.login.to_lowercase(),
+            repo.name.to_lowercase(),
+            default_pr_number()
+        )))
+        .respond_with(ResponseTemplate::new(200).set_body_json(PullRequest::default()))
+        .mount(mock_server)
+        .await;
+}
+
+#[derive(Serialize)]
+struct Head {
+    label: String,
+    #[serde(rename = "ref")]
+    ref_field: String,
+    sha: String,
+}
+
+#[derive(Serialize)]
+struct Base {
+    #[serde(rename = "ref")]
+    ref_field: String,
+    sha: String,
+}
+
+fn default_pr_number() -> u64 {
+    1
+}
+
+pub(crate) struct InMemoryIssue {
+    comments: Arc<Mutex<HashMap<u64, Vec<String>>>>,
+}
+
+impl InMemoryIssue {
+    pub(crate) async fn check_comments(&self, expected_comments: &[&str]) -> bool {
+        let pr_number = default_pr_number();
```

Comment:
This is quite hacky and could become flaky. What about creating a queue of comments and waiting until a comment is sent to it by a mocked impl?